app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -246,6 +250,15 @@ function FormSteps({
   const didSubmitWithErrors = useRef(false);
   const forceRender = useForceRender();
   const ifStillMounted = useIfStillMounted();
+  const [previousErrors, setPreviousErrors] = useState<PreviousErrorsLookup>({});
+
+  useEffect(() => {
+    const prevErrs = previousErrors[stepName || steps[0]?.name];
```

Comment:
Minor: It could be useful to disambiguate the two different "previous errors" here by incorporating "step" into the name, e.g.

```suggestion
    const previousStepErrors = previousErrors[stepName || steps[0]?.name];
```

Reply:
I decided to rename the state value instead of the one in `useEffect`. Given that the `const` in `useEffect` is used only on the following line and nowhere else, I don't think a rename of that variable is particularly helpful.